overlord/snapstate/snapstate_install_test.go

```diff
@@ -4020,3 +4020,42 @@ func (s *snapmgrTestSuite) TestHasAllContentAttributes(c *C) {
 	_, err = snapstate.HasAllContentAttrs(s.state, "some-snap", []string{"some"})
 	c.Assert(err.Error(), Equals, `expected 'content' attribute of slot 'bad-content-slot' (snap: 'some-snap') to be string but was int`)
 }
+
+func (s *snapmgrTestSuite) TestInstallPrereqIgnoreConfInSameChange(c *C) {
+	s.state.Lock()
+	snapstate.ReplaceStore(s.state, contentStore{fakeStore: s.fakeStore, state: s.state})
+
+	repo := interfaces.NewRepository()
+	ifacerepo.Replace(s.state, repo)
+
+	prodInfo := &snap.SideInfo{
+		RealName: "snap-content-slot",
+		SnapID:   "snap-content-slot-id",
+		Revision: snap.R(1),
+	}
+
+	chg := s.state.NewChange("install", "")
+
+	// To make the test deterministic, we inject a conflicting task to simulate
+	// an InstallMany({snap-content-plug, snap-content-slot}) with a failing snap-content-slot
+	conflTask := s.state.NewTask("conflicting-task", "test: conflicting task")
+	conflTask.Set("snap-setup", &snapstate.SnapSetup{SideInfo: prodInfo})
+	chg.AddTask(conflTask)
+
+	installTasks, err := snapstate.Install(context.Background(), s.state, "snap-content-plug", nil, s.user.ID, snapstate.Flags{})
+	c.Assert(err, IsNil)
+	c.Check(installTasks.Tasks(), Not(HasLen), 0)
+	chg.AddAll(installTasks)
+
+	s.state.Unlock()
+	// the tasks won't converge because of the conflicting-task
+	_ = s.o.Settle(testutil.HostScaledTimeout(3 * time.Second))
```

Comment:
Does this mean that this instruction will always take 3 seconds to complete? If so, shouldn't we make the timeout much smaller?

(this is what I assume from the "won't converge" comment above, at least)

Reply:
Yes, I wasn't sure of how small to make it while ensuring it's enough for it to pass even in constrained environments. wdyt, 1 second?

Reply:
I'd rather say 10 ms, but of course we should verify that's enough :-)

Unit tests should run quickly; if they don't, it means we need to add more mocking.

It seems weird, though, that we have an operation (Settle) which takes a timeout, but does not provide a channel for us to know when it's done...

Reply:
> It seems weird, though, that we have an operation (Settle) which takes a timeout, but does not provide a channel for us to know when it's done...

I don't understand this comment, Settle is synchronous, if it returns without errors it's done

Reply:
> > It seems weird, though, that we have an operation (Settle) which takes a timeout, but does not provide a channel for us to know when it's done...
> 
> I don't understand this comment, Settle is synchronous, if it returns without errors it's done

Maybe there's a misunderstanding from my side then: I asked if that instruction, in the context of the test, will always take three seconds to complete, and the answer was yes. Can you confirm that?

If it is so, why cannot the `Settle()` operation return as soon as it's either failed or completed?

Reply:
that's not true, the timeout it's a timeout, it will fail if it takes more than 3s, it doesn't mean it will always run for 3 sec

Reply:
Ah, OK, but then I'm fine with the current version of the code. :-)

Reply:
@mardy sorry, the comment I left above the settle call is wrong and mislead you. Just to clarify, I thought the conflicting task would prevent the task runner from converging because the task never stopped being in `DoStatus`. However, that's not true because the task has no handler so it's ignored by the task runner